### DISCUSSION TITLE
Update workflow to new version

### DISCUSSION THREAD

**Discussion by Georg632** [spartan-ng/spartan#517] [Q&A] [answered]
Hi,
I wanted to ask for some clarification about the recommended process for updating spartan/ui primitives to the newest version?
- Copy paste from the github repo?
- Rerun the cli (and reapply custom cva styles)?

Sorry if this is already explained in the documentation and I missed it!

(Thank you for the fantastic work!)

**Comment by goetzrobin** [comment, marked as answer]
I think the best way if you haven't made too many changes is to delete the spartan directory and remove the paths from the tsconfig and rerun the cli. but this is a main reason we are still in alpha because this process is very rough still

**Reply by Georg632** [reply]
Okay, that worked better than expected - thx!

I was a little bit confused by the brain imports. Is the intended way using `@spartan-ng/brain/<ui>` instead of alle the discrete packages?

**Comment by drdreo** [comment]
Ended up with the same flow. We have adapted the project in production (yes this is an alpha and we love to live on the edge) and migrating to latest versions basically weekly (toughest was when all imports were changed). My flow so far that worked is the delete flow. 

But perhaps we can add a migrator to the cli to make this simpler? 
I could imagine the exact same UI as with the ui generator that adds the packages in the first place, just cleaning all selected components up and re-installing them.

**Reply by Georg632** [reply]
Do you use the default spartan styling for your project? We are too using it in a production application but with an external designer, so we have to modify every hlm cva section to style it properly and add custom properties.
This adds a lot of comparison work when using the mentioned delete update flow, but it is doable.

Maybe it would be easier to apply the changes with a version comparison page, where you input your current spartan vesion and the version you want to update to.
The page shows all hlm-* files that have been modified. Then you can decide per file what you want to copy to your own vesion of the hlm file.
Does github provide this? I should be able to compare 2 production versions.
